### ISSUE TITLE
Auto-update app music list after "Update Audio Library"?

### ISSUE THREAD

**Issue by jeffvandyke** [xbmc/Kore#547]
So I've set up this app with a raspberry pi media center for my dear father, and he likes it very much, save for some frustrations. One is: we can add new music with Filezilla over the network, but then we have to update the music library (through Kore, library maintenance -> update audio library), but then we have to pull-down to refresh the music search results for the app to see the library's new music in Kore.

Since the app initiated the library refresh, could it automatically update it's local cache of music afterwards to see new results faster? That whole refresh process doesn't seem intuitive to seone who doesn't understand how caches work, and this might ease some frustrations.

Any thoughts?

**Comment by wonea** [comment]
I agree with this one, it's very frustrating to routinely have to refresh the content on my Kore apps, when I dump new music or video on the Kodi.

**Comment by axelsimon** [comment]
Also, this is needed to be able to achieve a fully automated system such as:
dump music to hard drive --> autotag and move to the right location (using something like [beets](http://beets.io/) --> update Kodi audio db (with a hook) --> update Kore audio db

**Comment by SyncedSynapse** [comment]
Makes sense, but implementing it would have some limitations. Basically, Kore must know when the library update on Kodi ends, and that is only transmitted to Kore if the connection is through TCP.

Anyway, i agree with this and it isn't hard, but be warned that it will only work in connections made through TCP.

**Comment by jeffvandyke** [comment]
Makes sense. I don't know the Kodi HTTP interface, though its [documented here](https://kodi.wiki/view/JSON-RPC_API/v9) I believe, but it would be nice if there was an endpoint to get the timestamp the library was last updated for Kore to check when trying to use the cached library (e.g., app startup on the music library page, switching to the music library). The video library is a very similar case too.

Oh, and other Kore remotes would still have to manually refresh if a different Kore app did the library refresh, so that's another good case for it.

**Comment by elcasa** [comment]
I agree that it's a problem, a normal user doesn't thinks that there are 2 different ways of refreshing data, one for the client & one for the server.

If it's not possible now, at least launch a toast message after the update audio library is sent, saying: 
_Remember to refresh your local library with a pull-down a the end!_

The message could be put also on searches that don't find results:
_Didn't found what you expect? Try refreshing audio library and then pull down to refresh local data_
